PoetAssistant/src/AppDelegate.swift

```diff
@@ -36,12 +37,50 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 		Settings.getTheme().apply()
 		return true
 	}
+	
+	func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
+		guard let sendingAppID = options[.sourceApplication] as? String else {
+			print("url from empty source")
+			return false
+		}
+		guard let ourMainBundleId = Bundle.main.bundleIdentifier else {
+			print("wtf, we don't know our own bundle id?")
+			return false
+		}
+		if !sendingAppID.hasPrefix(ourMainBundleId) {
+			print("url received from unknown app \(sendingAppID)")
+			return false
+		}
+		guard let components = NSURLComponents(url: url, resolvingAgainstBaseURL: true),
+			let queryPath = components.path, queryPath.starts(with: "query/") else {
```

Comment:
Should probably check against our known exact app extensions, instead of a prefix check.